agent/config/config.go

```diff
@@ -249,7 +252,9 @@ func environmentConfig() Config {
 	// invalid parse
 	// Blank is not a warning; we have sane defaults
 	if err != io.EOF && err != nil {
-		seelog.Warnf("Invalid format for \"ECS_RESERVED_PORTS_UDP\" environment variable; expected a JSON array like [1,2,3]. err %v", err)
+		msg := fmt.Sprintf("Invalid format for \"ECS_RESERVED_PORTS_UDP\" environment variable; expected a JSON array like [1,2,3]. err %v", err)
+		seelog.Warn(msg)
+		errs = append(errs, errors.New(string(msg)))
```

Comment:
This and other places where you append to `errs` apart from when parsing `ECS_INSTANCE_ATTRIBUTES` means that we are introducing a breaking change in the way env config works, correct? While I agree that its good to fail early rather then letting instances to be registered where they might be missing certain attributes, it also means that customers who have launch configs and ASGs today with wrong config will suddenly find that their instances do not show up anymore after upgrading the agent. I feel uncomfortable pushing this breaking change. would also like to get @cbbarclay's thoughts on this.